core/modules/entity/entity.module

```diff
@@ -140,6 +140,34 @@ function entity_view_mode_exists($machine_name, $element, $form_state) {
   return !empty($entity_info['view modes'][$machine_name]);
 }
 
+/**
+ * Determines whether the given user has access to an entity.
+ *
+ * @param string $op
+ *   The operation being performed. One of 'view', 'update' or 'delete'.
+ * @param string $entity_type
+ *   The entity type of the entity to check for.
+ * @param EntityInterface $entity
+ *   Optionally an entity to check access for. If no entity is given, it will be
+ *   determined whether access is allowed for all entities of the given type.
+ * @param User $account
+ *   The user to check for. Leave it to NULL to check for the global user.
+ *
+ * @return bool
+ *   Whether access is allowed or not. If the entity type does not specify any
+ *   access information, NULL is returned.
+ *
+ * @see entity_type_supports()
+ * @deprecated since 1.0.0
+ */
+function entity_access($op, $entity_type, EntityInterface $entity = NULL, User $account = NULL) {
+  if (!in_array($op, array('view', 'update', 'delete'), TRUE)) {
+    // If the $op was not one of the supported ones, we return access denied.
+    return FALSE;
+  }
+  return $entity->access($op, $account);
+}
```

Comment:
How do you feel about supporting `entity_access('create', ...)`?

Reply:
Also see why in D8 they use access() and createAccess()
https://www.drupal.org/docs/8/api/entity-api/working-with-the-entity-api

"When checking create access, there is usually no entity yet. Creating one just to check if someone would be able to create it is a costly operation. Therefore, create access for those should be checked directly on the access controller."

However, it seems they do allow create for Entity::access().

"If there is already an entity, $entity->access('create') works too, which just forwards to the createAccess() method, the same way other operations forward to the access() method on the access controller."

Reply:
> What would be the main reason for allowing entity_access('create', ...)? Backward compatibility or something else?

I could see modules needing to generically check create access, in the same way they need to check update or delete access. Like `node_access()`, instead of passing the entire object you'd only be given the bundle type (if one exists for that entity type).

I believe generally the code would look like this:

```
if ($op === 'create') {
  $info = entity_get_info($entity_type);
  return $info['class']::createAccess($entity, $account);
}
else {
  return $entity->access($op, $account);
}
```

Not totally sure that syntax is perfect for calling a static method on a dynamic class name though.